src/intersectionWith.js

```diff
@@ -12,7 +12,7 @@ var uniqWith = require('./uniqWith');
  * @memberOf R
  * @since v0.1.0
  * @category Relation
- * @sig (a -> a -> Boolean) -> [*] -> [*] -> [*]
+ * @sig ((a, a) -> Boolean) -> [a] -> [a] -> [a]
```

Comment:
I'd forgotten that #1597 introduced a regression which means that the more general signature proposed above no longer works reliably:

```javascript
> R.intersectionWith(R.prop('x'), [{x: 1, y: 1}], [{x: 1}])
[{x: 1, y: 1}]

> R.intersectionWith(R.prop('x'), [{x: 1}], [{x: 1, y: 1}])
[{x: 1}]
```

I'll open a pull request to address this. We introduced a regression. We should fix the regression *then* discuss the value of replacing a more general function with a less general one.